### PR TITLE
Drop support for AWS credentials using env vars and java props & enable s3.client.default.access/secret_key settings

### PR DESCRIPTION
**Pull request by matriv**
Please check commit messages.

### CODE REVIEW

File: blackbox/docs/sql/statements/copy-from.rst [hunk 1]
```diff
@@ -130,13 +130,6 @@ Can be used to access buckets on the Amazon AWS S3 Service:
 
     s3://[<accesskey>:<secretkey>@]<bucketname>/<path>
 
-If ``accesskey`` and ``secretkey`` are ommited, CrateDB attempts to load the
```
Comment (seut) [inline blackbox/docs/sql/statements/copy-from.rst]: should we change that to also note the fallback to config file parameters?

File: blackbox/docs/sql/statements/copy-to.rst [hunk 2]
```diff
@@ -75,11 +75,6 @@ The resulting string should be a valid URI of one of the supporting schemes:
 If no scheme is given (e.g.: '/path/to/dir') the default uri-scheme ``file://``
 will be used.
 
-If the s3 scheme is used without specifying any credentials an attempt is made
```
Comment (seut) [inline blackbox/docs/sql/statements/copy-to.rst]: same here

File: blackbox/docs/sql/statements/create-repository.rst [hunk 3]
```diff
@@ -230,25 +230,25 @@ A repository that stores its snapshot on the Amazon S3 service.
 
 **access_key**
   | *Type:*    ``string``
-  | *Default:* Globally defined environmental variable ``AWS_ACCESS_KEY_ID``.
```
Comment (seut) [inline blackbox/docs/sql/statements/create-repository.rst]: the default is now: config file parameter

File: blackbox/docs/sql/statements/create-repository.rst [hunk 4]
```diff
@@ -230,25 +230,25 @@ A repository that stores its snapshot on the Amazon S3 service.
...
 
-     If the access key is set explicity (not via environment variable) it will
-     be visible in plain text when querying the ``sys.repositories`` table.
+     If the secret key is set explicitly (not via :ref:`configuration setting
+     <s3-credentials-access-key>`) it will be visible in plain text when
+     querying the ``sys.repositories`` table.
 
 **secret_key**
   | *Type:*    ``string``
-  | *Default:* Globally defined environmental variable ``AWS_SECRET_KEY``.
```
Comment (seut) [inline blackbox/docs/sql/statements/create-repository.rst]: also here

File: blackbox/docs/sql/statements/copy-from.rst (hunk 1 above)
Comment (matriv) [reply blackbox/docs/sql/statements/copy-from.rst]: Nope..... these params are only for CREATE REPOSITORY. Copy from/to are once off SQL statements, no credentials are exposed through sys table.

File: blackbox/docs/sql/statements/copy-to.rst (hunk 2 above)
Comment (matriv) [reply blackbox/docs/sql/statements/copy-to.rst]: same here ;)

**Comment by matriv** [comment]
@smu Thx! pushed fixup.

File: blackbox/docs/sql/statements/copy-from.rst (hunk 1 above)
Comment (seut) [reply blackbox/docs/sql/statements/copy-from.rst]: ah right :)

**Comment by matriv** [comment]
Backported to 3.0 with https://github.com/crate/crate/commit/c3df1b47f9358c3acef00b99c3a9e2625c2946e6 and https://github.com/crate/crate/commit/e7a6088da9c37c716c495bcb5edcd17b9b3d5588

**Pull request merged**